### DISCUSSION TITLE
OEmbed property editor in core

### DISCUSSION THREAD

**Discussion by bjarnef** [umbraco/Umbraco-CMS#15038] [Features and ideas]
We use the OEmbed property editor is most projects and includes it in our starter project:
https://marketplace.umbraco.com/package/dawoe.oembedpickerpropertyeditor

which allow single/multiple selection of OEmbed.

Basically it is just wrapping the existing embed infinite editor in core.
https://github.com/umbraco/Umbraco-CMS/tree/b423329d978a75aebd28124dea7c2137cd6beb1f/src/Umbraco.Web.UI.Client/src/views/common/infiniteeditors/embed

I think this would also be useful to support on Umbraco Heartcore.

**Comment by nul800sebastiaan** [comment]
Speaking personally, not on behalf of HQ: I wish anything to do with OEmbed was a package. We've seen over the years that endpoints and configurations change or.. sadly even die. To be able to update these externally without having to upgrade to a new Umbraco version is would be very valuable. 
So instead of moving this into core, my suggestion would be to have an external package that deals with all things OEmbed.

Alternatively, being able to change the OEmbed providers config without having to use code (so a config) might fix the problem with providers changing/disappearing.

I might be wrong though, and if all of this is already possible then ignore me 😉

**Reply by bjarnef** [reply]
I think it could make sense to have a integration project to handle integration with OEmbed.
https://github.com/umbraco/Umbraco.Cms.Integrations/tree/main/src

Maybe some default providers or each own as a separate package? or some additional packages for some less common providers.

E.g.

- Umbraco.Cms.Integrations.OEmbed.Core
- Umbraco.Cms.Integrations.OEmbed.YouTube
- Umbraco.Cms.Integrations.OEmbed.Vimeo

**Comment by bjarnef** [comment]
@AndyButland @acoumb with the integrations for Umbraco, I wonder if OEmbed should be extracted from Umbraco CMS core?
https://github.com/umbraco/Umbraco.Cms.Integrations

Maybe obsolete in v14 and removed in v15 or v16?
We use this all the time to select OEmbed in a property, but basically just wrapping what's already in core (e.g. via richtext editor), but not a standalone property editor:
https://marketplace.umbraco.com/package/dawoe.oembedpickerpropertyeditor

Not sure it there are other alternatives out there, but potentially it could allow replacing OEmbed with something else?

**Comment by bjarnef** [comment]
@sajumb I think this could potentially save some (maybe a lot) resources on Umbraco Cloud as well?
Not all projects need OEmbed or only a few of the OEmbed providers.
Mostly of our clients only use YouTube and Vimeo.

Some agencies also implements there own implementations instead like @abjerner https://marketplace.umbraco.com/search/skybrud

I think it would be great if with a OEmbed core package that installed plugin for TinyMCE and a property editor + addional providers and ideally replacing this package.
https://marketplace.umbraco.com/package/dawoe.oembedpickerpropertyeditor

I see this a bit similar to Payment providers in Umbraco Commerce (and upcoming Shipping providers).
